### PR TITLE
feat(vscode): add command "Copy To Clipboard Without Brackets 

### PR DESCRIPTION
**Pull request by jsjoeio**
This adds a new command for `foam-vscode`  called "Copy To Clipboard Without Brackets". I discussed this in the Foam Discord a [while back](https://discordapp.com/channels/729975036148056075/730056943997288540/739321725967138847), asking if this should be an outside thing (separate extension) or added as a feature to the core extension. It was suggested to do the later. 

## Use Case

Sometimes when drafting messages to people in Foam, I use references. Instead of sending them the messages with the brackets or removing them, I thought it would be nice to have this command which copies to clipboard and removes them for you.

### CODE REVIEW

File: packages/foam-vscode/src/utils.ts [hunk 1]
```diff
@@ -91,3 +91,35 @@ export function dropExtension(path: string): string {
...
+ * Used for the "Copy to Clipboard Without Brackets" command
+ *
+ */
+export function removeBrackets(s: string): string {
+  // take in the string, split on space
+  const stringSplitBySpace = s.split(" ");
+
+  // loop through words
+  const modifiedWords = stringSplitBySpace.map(currentWord => {
+    if (currentWord.includes("[[")) {
```
Comment (riccardoferretti) [inline packages/foam-vscode/src/utils.ts:105]: Are we happy with not supporting spaces in the file name/path?
(Even not taking into consideration aliases which we don't support yet)

File: packages/foam-vscode/src/utils.ts [hunk 2]
```diff
@@ -91,3 +91,35 @@ export function dropExtension(path: string): string {
...
+      let word = currentWord.replace(/(\[\[)/g, "");
+      word = word.replace(/(\]\])/g, "");
+      word = word.replace(/(.mdx|.md|.markdown)/g, "");
+      word = word.replace(/[-]/g, " ");
+
+      // now capitalize every word
+      const modifiedWord = word
```
Comment (riccardoferretti) [inline packages/foam-vscode/src/utils.ts:112]: Basically are we "deslugifying" here?

File: packages/foam-vscode/src/utils.ts [hunk 3]
```diff
@@ -91,3 +91,35 @@ export function dropExtension(path: string): string {
...
+export function removeBrackets(s: string): string {
+  // take in the string, split on space
+  const stringSplitBySpace = s.split(" ");
+
+  // loop through words
+  const modifiedWords = stringSplitBySpace.map(currentWord => {
+    if (currentWord.includes("[[")) {
+      let word = currentWord.replace(/(\[\[)/g, "");
+      word = word.replace(/(\]\])/g, "");
+      word = word.replace(/(.mdx|.md|.markdown)/g, "");
```
Comment (riccardoferretti) [inline packages/foam-vscode/src/utils.ts:108]: Do we have this list somewhere already? Should we consolidate it or do you think it's better to keep things more decoupled at the moment?

**Review by riccardoferretti — APPROVED** [review]
This looks really good to me. 
I have only left a couple of comments around some edge cases or minor points.

File: packages/foam-vscode/test/utils.test.ts [hunk 4]
```diff
@@ -1,10 +1,49 @@
...
   test("returns file name without extension", () => {
     expect(dropExtension('file.md')).toEqual('file');
   });
 });
+
+describe("removeBrackets", () => {
+  it("removes the brackets", () => {
+    const input = "hello world [[this-is-it]]";
+    const actual = removeBrackets(input);
+    const expected = "hello world This Is It";
```
Comment (ingalless) [inline packages/foam-vscode/test/utils.test.ts:16]: Perhaps it isn't for this PR, but I sometimes name files differently to my heading, so I would want to grab the title. Is that an easy change? No worries if not but just floating it...
Like the tests though!

**Review by ingalless — APPROVED** [review]
This looks great! Very useful command

File: packages/foam-vscode/test/utils.test.ts (hunk 4 above)
Comment (riccardoferretti) [reply packages/foam-vscode/test/utils.test.ts:16]: That's actually a very good point, especially if we already have the definitions. And will probably leverage more the foam metadata as opposed to simply the markdown itself

File: packages/foam-vscode/src/utils.ts (hunk 3 above)
Comment (jsjoeio) [reply packages/foam-vscode/src/utils.ts:111]: Good call! I didn't even think to check. I'll look and if it is, I'll use that

File: packages/foam-vscode/src/utils.ts (hunk 2 above)
Comment (jsjoeio) [reply packages/foam-vscode/src/utils.ts]: Yes! Exactly! I can update the comments to make that more obvious. Happy to even break it into a small function called `deslugify`

File: packages/foam-vscode/test/utils.test.ts (hunk 4 above)
Comment (jsjoeio) [reply packages/foam-vscode/test/utils.test.ts:16]: Good point! I didn't even think about. 

>  Is that an easy change?

I don't know 😅 More than happy to accept change if one of y'all want to do it!

File: packages/foam-vscode/src/utils.ts (hunk 3 above)
Comment (jsjoeio) [reply packages/foam-vscode/src/utils.ts:111]: Okay I found it inside `docs/_layouts/foam.html`, but can't export it because it's just in a script tag in an HTML file 🤔

I'm not seeing it elsewhere...I'm going to leave for now, but if you see a better option, let me know and I'll happily change it!

File: packages/foam-vscode/src/utils.ts (hunk 2 above)
Comment (jsjoeio) [reply packages/foam-vscode/src/utils.ts]: Actually...it title cases the word, so I broke it out into a function, added more comments and a few tests. Good callout!

**Review by jsjoeio — COMMENTED** [review]
Thanks for the specific feedback @ingalless and @riccardoferretti ! 🙌🏼

I responded and made a few changes. Feel free to take another look! If there's changes to be made, hit the "request changes". If it's ready to go by chance, feel free to merge!

**Comment by jsjoeio** [comment]
Chatted over Discord - good to merge!
